kebechet/managers/update.py

```diff
@@ -508,9 +522,9 @@ def _do_update(self, labels: list, pipenv_used: bool = False) -> dict:
             else:
                 # We were able to update all, close reported issue if any.
                 self.sm.close_issue_if_exists(_ISSUE_UPDATE_ALL_NAME, comment=ISSUE_CLOSE_COMMENT.format(sha=self.sha))
-        else:  # requirements.txt
-            old_environment = self._get_requirements_txt_dependencies()
-            direct_dependencies = self._get_direct_dependencies_requirements()
+        else:  # either requirements.txt or requirements-dev.txt
+            old_environment = self._get_requirements_txt_dependencies(req_dev)
+            direct_dependencies = self._get_direct_dependencies_requirements(req_dev)
             old_direct_dependencies_version = {k: v for k, v in old_environment.items() if k in direct_dependencies}
```

Comment:
@fridex while processing the requirements.txt first, there is no problem for direct dependencies but while processing the requirements-dev.txt after requirements.txt. I'm getting the error : `f"Failed to retrieve version information for dependency {dependency}, (dev: {is_dev})")
kebechet.exception.InternalError: Failed to retrieve version information for dependency factory_boy, (dev: False)`

please let me know where i m going wrong. :)